pkg/server/server.go

```diff
@@ -354,6 +401,17 @@ func (s *Server) Check(ctx context.Context, req *openfgav1.CheckRequest) (*openf
 		graph.WithMaxConcurrentReads(s.maxConcurrentReadsForCheck),
 	)
 
+	if s.checkCache != nil {
+		cachedCheckResolver := graph.NewCachedCheckResolver(
+			checkResolver,
+			graph.WithExistingCache(s.checkCache),
+			graph.WithCacheTTL(s.checkQueryCacheTTL),
+		)
+
+		checkResolver.SetDelegate(cachedCheckResolver)
+		defer cachedCheckResolver.Close()
+	}
+
 	resp, err := checkResolver.ResolveCheck(ctx, &graph.ResolveCheckRequest{
```

Comment:
We're not using the `cachedCheckResolver` here. We should be. Right now nothing is being cached.

Reply:
I suggest the following

```go
var resolver graph.CheckResolver

checkResolver := graph.NewLocalCheck(...)
resolver = checkResolver

if s.checkCache != nil {
  cachedCheckResolver := graph.NewCachedCheckResolver(...)

  checkResolver.SetDelegate(cachedCheckResolver)
  resolver = cachedCheckResolver
}

resp, err := resolver.ResolveCheck(...)
```

Reply:
I don't understand how that code is different to adrian's code

Reply:
Added test + added in changes as suggested by @jon-whit .  The code is different in that the initial call to ResolveCheck should be through cached check resolver.